src/streamlink/plugins/nbcnews.py

```diff
@@ -67,8 +44,16 @@ class NBCNews(Plugin):
     def get_title(self):
         return 'NBC News Now'
 
+    def _get_video_id(self, site_url):
+        maininfo_res = self.session.http.get(site_url)
+        video_info_json = re.findall(r'<script type="application/ld\+json">(.*?)</script>', maininfo_res.text)[0]
+        video_info_json = parse_json(video_info_json)
+        embedUrl = video_info_json["embedUrl"]
+        video_id = embedUrl.split("/")[-1]
+        return video_id
```

Comment:
This should be done in the validation schema, just like before, because it properly handles non-matching content via `validate.any(None, ...)` and thus doesn't raise any exceptions. You probably also meant to use the regex from the plugin's class attribute.

```py
    json_data_schema = validate.Schema(
        validate.transform(json_data_re.search),
        validate.any(None, validate.all(
            validate.get(1),
            validate.transform(parse_json),
            {"embedUrl": validate.url()},
            validate.get("embedUrl"),
            validation.transform(lambda url: url.split("/")[-1])
        ))
    )
```

The plugin's `_get_streams` should ideally also check if `video_id is None` and return in this case, so that it doesn't continue with missing data.

```py
    video_id = self.session.http.get(self.url, schema=self.json_data_schema)
    if video_id is None:
        return
```